src/net/tcp.rs

```diff
@@ -51,6 +51,9 @@ use futures::task::{Context, Poll};
 /// use futures::prelude::*;
 /// use runtime::net::TcpStream;
 ///
+/// # #[cfg(not(feature = "native"))]
+/// # fn main() { }
+/// # #[cfg(feature = "native")]
```

Comment:
Perhaps a better way of doing this would be to disable the doc tests on CI when running without the `native` feature.

I'm not super keen on adding this line to _every_ example; ideally we could work around test failure in a different way.

Reply:
I can't find other ways yet to ignore the doc test on the specific feature without setting on every doc test.

So we have two option:
- Specifying feature on every doc test (with `/// #[cfg(...)]` or `#[cfg_attr(..., doc = ...)]` or something like that)
- State that doc tests must be run with `native` feature

Reply:
Hmm, neither of these options seems great. And doesn't provide any extra guarantees on tests that we might want. I'm kind of thinking that perhaps we should not bother with making our tests work without default features.

I feel it seems like a better approach than individually having to turn off each and every doc test we have.

Reply:
We might have to leave the doc test broken and wait for some other features (such as `doc_cfg`) to be stable.

Reply:
Sounds good!